clang/lib/CodeGen/Targets/AMDGPU.cpp

```diff
@@ -52,6 +52,12 @@ class AMDGPUABIInfo final : public DefaultABIInfo {
   void computeInfo(CGFunctionInfo &FI) const override;
   RValue EmitVAArg(CodeGenFunction &CGF, Address VAListAddr, QualType Ty,
                    AggValueSlot Slot) const override;
+
+  llvm::FixedVectorType *
+  getOptimalVectorType(llvm::FixedVectorType *T,
+                       const LangOptions &) const override {
+    return T;
```

Comment:
We only want to preserve this for the 96-bit cases where we have legal instructions

Reply:
With a direct instruction, yes. We can't widen the stores in the other odd cases so we need the hint that it's OK by using the wider type